spec/apps/dummy/config/initializers/scimitar.rb

```diff
@@ -27,6 +27,10 @@ def scim_schemas_url(options)
       def scim_resource_type_url(options)
         super(test: 1, **options)
       end
+
+      def valid_token
+        'A'
+      end
```

Comment:
I tried to put this directly in the test file

```ruby
    before do
      Scimitar.engine_configuration = Scimitar::EngineConfiguration.new(
        application_controller_mixin: Module.new do
          def valid_token
            'A'
          end
        end,
        token_authenticator: Proc.new do | token, options |
          token == valid_token
        end
      )
    end
```

but it doesn't work, I don't know why.

Reply:
Spent a while looking into this on your branch.

The file you're altering is for the 'dummy' app embedded in the test suite for `request` type specs. The tests you added are for the base `Scimitar::ApplicationController` as a unit test, so don't include any such configuration. However, it looks like I did not account for resetting Scimitar configuration in the test suite, so there's pollution of the configuration. It starts off essentially at 'default', then as soon as one of the rest specs runs, picks up the file here and goes with that configuration thereafter.

You see more test passes that you might because you're still using `A` as a valid token here, which is used in other tests too and can give a false sense of security on whether or not things pass! If `spec/controllers/scimitar/application_controller_spec.rb` runs the context at line 64 before your own tests on an otherwise-default configuration, the leaky configuration change there would cause your tests, if executed subsequently, to pass - but your configuration is not being exercised. The way to avoid false-positives here is simple - changing to something else, such as `B`, for the dummy application's `#valid_token`.

Before the PR can proceed further, I need to do some work to figure out an appropriate way to reset configuration changes between tests, while making sure that configuration *is* asserted as expected during e.g. request specs. Once that's done, it should become clearer what goes where.

Reply:
Ah I see, I didn't bother to check that `Scimitar.engine_configuration=` method is memorized

https://github.com/RIPAGlobal/scimitar/blob/3394c6a3bb2699bad581f3d453aa6f705ce7b259/lib/scimitar.rb#L19-L23

Make total sense!

I've made the changes [`065c4dd` (#88)](https://github.com/RIPAGlobal/scimitar/pull/88/commits/065c4dd1c0fea21e3631c4d94b9b278f571be783)

Let me know if I'm still missing something! Thanks!